techtv2ovs/utils.py

```diff
@@ -0,0 +1,302 @@
+                ttv_video.thumbnail_status = ImportStatus.ERROR
+                ttv_video.errors += '{}\n\n'.format(str(exc))
+
+        if ttv_video.thumbnail_status != ImportStatus.ERROR:
+            ttv_video.thumbnail_status = ImportStatus.COMPLETE
+        ttv_video.save()
+
+    def process_files(self, ttv_video, s3videos):
+        """
+        Import videofiles, thumbnails, and captions for a TechTVVideo.
+
+        Args:
+            ttv_video (TechTVVideo): The TechTVVideo object to process
+            s3videos (list): List of video file S3 objects
+        """
+        self.process_thumbs(ttv_video)
+        process_videofiles.delay(ttv_video.id, s3videos, self.aws)
+
+    def process_videos(self, ttvcollection):
+        """
+        Process all the videos for a TechTV Collection
+
+        Args:
+            ttvcollection (TechTVCollection): a TechTVCollection object
+        """
+        video_query = (
+            'SELECT v.id, v.title, v.description, v.external_id, v.private, v.private_token '
+            'from videos v inner join collection_videos cv on cv.video_id = v.id where cv.collection_id = {} '
+            'and v.status = \'approved\' AND v.file_restored = 1 AND v.locked=0;'.format(ttvcollection.id)
+        )
+        videos = mysql_query(video_query, self.connection)
+        for video in videos:
+            (v_id, v_title, v_description, external_id, private, token) = video
+            self.output.write('Processing video: {}'.format(v_title))
+            ttvvideo, _ = TechTVVideo.objects.update_or_create(
+                ttv_id=v_id,
+                ttv_collection=ttvcollection,
+                defaults={
+                    'title': v_title,
+                    'description': v_description,
+                    'external_id': external_id,
+                    'private': True if private == 1 else False,
+                    'private_token': token
+                }
+            )
+            s3videos = get_s3_videos(external_id)
+            if not s3videos:
+                ttvvideo.status = ImportStatus.MISSING
+                ttvvideo.videofile_status = ImportStatus.MISSING
+                ttvvideo.save()
+                continue
+            if not ttvvideo.video:
+                ttvvideo.video = Video.objects.create(
+                    collection=ttvcollection.collection,
+                    title=v_title,
+                    description=remove_tags(v_description),
+                    source_url="http://techtv.mit.edu/videos/",
+                    is_private=ttvvideo.private,
+                )
+            ttvvideo.status = ImportStatus.CREATED
+            ttvvideo.save()
+            self.process_files(ttvvideo, s3videos)
+
+    def process_collection(self, *collection):
+        """
+        Iterate over every TechTV collection
+
+        Args:
+            collection (tuple): Tuple (row) of collection data from MySQL
+        """
+        (cid, name, description, email) = collection
```

Comment:
Can we move these arguments into the function arguments instead of doing tuple unpacking here?

Reply:
✅